Library/Homebrew/test/utils/tty_spec.rb

```diff
@@ -0,0 +1,54 @@
+require "utils"
+
+describe Tty do
+  specify "::strip_ansi" do
+    expect(described_class.strip_ansi("\033\[36;7mhello\033\[0m")).to eq("hello")
+  end
+
+  specify "::width" do
+    expect(described_class.width).to be_kind_of(Integer)
+    expect(described_class.width).to be >= 0
+  end
+
+  describe "::truncate" do
+    it "truncates the text to the screen width" do
+      allow(described_class).to receive(:width).and_return(15)
+
+      expect(described_class.truncate("foobar something very long")).to eq("foobar some")
+      expect(described_class.truncate("truncate")).to eq("truncate")
+    end
+
+    it "doesn't truncate the text if the terminal is unsupported" do
+      allow(described_class).to receive(:width).and_return(0)
+      expect(described_class.truncate("foobar something very long")).to eq("foobar something very long")
+    end
+  end
+
+  context "$stdout is not a TTY" do
+    it "returns an empty string for all colors" do
+      allow($stdout).to receive(:tty?).and_return(false)
```

Comment:
This should be in a `before` block since it's doing what's described in the context.